scripts/categorizer.py

```diff
@@ -385,14 +412,61 @@ def main():
         is_conditional = determine_if_conditional(line_number, multiline_number, line, categorizations) # This might be an exclusive one
         is_function_def = determine_if_function_def(line_number, multiline_number, line, categorizations)
         is_assignment = determine_if_assignment(line_number, multiline_number, line, categorizations, is_function_def)
```

Comment:
There is an implicit relationship between assignments -> function_defs, loops, and imports. The function_defs, loops, and imports are all assignments. So I may decide to make assignments just be those `variable=value` rather than include all of them. I was hoping to avoid relationships at this level between categorizations... Think on this.

Reply:
Also if there is an relationship then it maters in which order these functions are called.

Reply:
multiline is already a relationship that I have to manage between lines. I don't know how I feel about categorizing in a way that has hierarchy relationships of one belonging to another.

Reply:
What do you mean by a relationship between assignments? For example, if I declared a variable and later used it, would those two lines be considered as related?

Reply:
I'm referring to the fact that I am making a link between two different types of categorizations like for_loops and assignments. For_loops normally happen to define an iterator. This is an assignment if I use the fact in code that a for loop can always be considered an assignment. I am implicitly saying that for-loops are a sub-category of assignments. I might also want to later use the fact that for-loops are a loop when/if I categorize loops. 

I came into the project with the design philosophy that I all I wanted to do with this portion is to identify something. I did not want to establish relationships between any categorizations. Using the relationship in the actual code seems the right thing to do as it reduces redundant logic, but I am making sure that I know the cost is now that I will be creating a dependency between assignment and for loop categorizations which now could mean that a potential bug could either be in one of the sub-categories or the assignment category. It could also be in the multiline categorization. I'm making skyscrapers out of a garden shack in order to try to increase my awareness of what I'm doing when designing or creating something.

What will really make this either important or not important will be use case, hence why it is on my todo list before I categorizing too many more new categories.